src/docfx/lib/schema/JsonSchemaValidation.cs

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Docs.Build
+{
+    internal static class JsonSchemaValidation
+    {
+        public static List<Error> Validate(JsonSchema schema, JToken token)
+        {
+            var errors = new List<Error>();
+            ValidateCore(schema, token, errors);
+            return errors;
+        }
+
+        private static void ValidateCore(JsonSchema schema, JToken token, List<Error> errors)
+        {
+            if (!TypeMatches(schema.Type, token.Type))
+            {
+                errors.Add(Errors.UnexpectedType(JsonUtility.GetSourceInfo(token), schema.Type.ToString(), token.Type.ToString()));
+                return;
+            }
+
+            switch (token)
+            {
+                case JValue scalar:
+                    if (schema.Enum.Count > 0 && !schema.Enum.Contains(scalar))
+                    {
+                        errors.Add(Errors.UndefinedValue(JsonUtility.GetSourceInfo(token), scalar, schema.Enum));
+                    }
+                    break;
+
+                case JArray array:
+                    if (schema.Items != null)
+                    {
+                        foreach (var item in array)
+                        {
+                            ValidateCore(schema.Items, item, errors);
+                        }
+                    }
+                    break;
+
+                case JObject map:
+                    foreach (var (key, value) in map)
+                    {
+                        if (schema.Properties.TryGetValue(key, out var propertySchema))
```

Comment:
How to handle `AdditionalProperties` is false or object?
https://json-schema.org/understanding-json-schema/reference/object.html#properties

Reply:
Defer that to SDP work

Reply:
Support it during SDP implementation